core/date/suggest.go

```diff
@@ -18,55 +18,69 @@ import (
 	"github.com/anyproto/anytype-heart/pkg/lib/localstore/objectstore"
 	"github.com/anyproto/anytype-heart/space"
 	"github.com/anyproto/anytype-heart/util/dateutil"
+	"github.com/anyproto/anytype-heart/util/pbtypes"
 )
 
-func EnrichRecordsWithDateSuggestion(
+func EnrichRecordsWithDateSuggestions(
 	ctx context.Context,
 	records []database.Record,
 	req *pb.RpcObjectSearchRequest,
 	store objectstore.ObjectStore,
 	spaceService space.Service,
 ) ([]database.Record, error) {
-	dt := suggestDateForSearch(time.Now(), req.FullText)
-	if dt.IsZero() {
+	ids := suggestDateObjectIds(req)
+	if ids == nil {
```

Comment:
It is not a good practice to compare slice with nil, cause it can be empty but not nil. In your case if client sends id filter but it has no date ids suggestDateObjectIds will return empty slice instead of nil.

Reply:
Yeah, thank you for mentioning that. Fixed!